src/cloudformation_cli_python_lib/metrics.py

```diff
@@ -0,0 +1,96 @@
+import logging
+from typing import Any, Mapping
+
+# boto3 doesn't have stub files
+from boto3.session import Session  # type: ignore
+
+from botocore.exceptions import ClientError  # type: ignore
+
+from .interface import Action, MetricTypes, StandardUnit
+
+LOG = logging.getLogger(__name__)
+
+
+def format_dimensions(dimensions: Mapping[Any, Any]) -> Any:  # TODO fix type
+    formatted_dimensions = []
+    for key, value in dimensions.items():
+        formatted_dimensions.append({"Name": key, "Value": value})
+    return formatted_dimensions
+
+
+class MetricPublisher:
+    def __init__(self, namespace: str, session: Session) -> None:
+        self.namespace = namespace
+        self.client = session.client("cloudwatch")
+
+    def _publish_metric(  # pylint: disable-msg=too-many-arguments
+        self,
+        metric_name: MetricTypes,
+        dimensions: Mapping[Any, Any],  # TODO: fix type
+        unit: StandardUnit,
+        value: Any,
+        date: float,
+    ) -> None:
+        dimensions = format_dimensions(dimensions)
+        try:
+            res = self.client.put_metric_data(
+                Namespace=self.namespace,
+                MetricData=[
+                    {
+                        "MetricName": metric_name,
+                        "Dimensions": dimensions,
+                        "Unit": unit,
+                        "Timestamp": date,
+                        "Value": value,
+                    }
+                ],
+            )
+            LOG.debug(res)
```

Comment:
does this help resource authors? bit worried about log spam

Reply:
put_metric_data actually returns None so i've removed the log.